include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -1004,7 +1004,7 @@ namespace VectorTools
       Assert (dof.get_fe().degree == fe_degree,
               ExcDimensionMismatch(fe_degree, dof.get_fe().degree));
 
-      typedef MatrixFreeOperators::MassOperator<dim, fe_degree, 1, Number> MatrixType;
+      typedef MatrixFreeOperators::MassOperator<dim, fe_degree, n_q_points_1d, 1, Number> MatrixType;
```

Comment:
whereas here the projection is from MatrixFree quadrature data with specified number of quadrature points. Use the same for MassOperator.